app/src/bat_charger.cpp

```diff
@@ -205,42 +205,39 @@ void battery_conf_init(BatConf *bat, int type, int num_cells, float nominal_capa
 }
 
 // checks settings in bat_conf for plausibility
+#include <functional>
 bool battery_conf_check(BatConf *bat_conf)
 {
     // things to check:
     // - load_disconnect/reconnect hysteresis makes sense?
     // - cutoff current not extremely low/high
     // - capacity plausible
+    //
+
+  	struct condition { std::function<bool()> func; const char* text;};
+
+	const condition conditions[] = {
+	{ .func = [bat_conf]() { return bat_conf->load_reconnect_voltage > (bat_conf->load_disconnect_voltage + 0.4); }, .text = "Load Reconnect Voltage must be higher than Load Disconnect Voltage + 0.4" },
```

Comment:
Lines too long (should be <100 chars) and tabs are used instead of 4 spaces for indentation.